File: cpp/PeripheralDevice.cpp
// PeripheralDevice.cpp
#pragma once
#include <iostream>
#include <string>
using namespace std;

class PeripheralDevice {
protected:
    string interfaceType;
    bool isWireless;
    
public:
    PeripheralDevice() {}
    
    PeripheralDevice(string interfaceType, bool isWireless) {
        this->interfaceType = interfaceType;
        this->isWireless = isWireless;
    }
    
    void setInterfaceType(string interfaceType) {
        this->interfaceType = interfaceType;
    }
    
    void setWireless(bool isWireless) {
        this->isWireless = isWireless;
    }
    
    string getInterfaceType() {
        return this->interfaceType;
    }
    
    bool getIsWireless() {
        return this->isWireless;
    }
    
    ~PeripheralDevice() {}
};

File: python/PeripheralDevice.py
class PeripheralDevice :
    def __init__(self, interfaceType, isWireless):
        
        self.__interfaceType = interfaceType
        self.__isWireless = isWireless
        
    def get_interfaceType(self):
        return self.__interfaceType
    def get_isWireless(self):
        return self.__isWireless
    
    # Setter methods
    def set_interfaceType(self, interfaceType):
        self.__interfaceType = interfaceType
    def set_isWireless(self, isWireless):
        self.__isWireless = isWireless